lib/TopSitesFeed.jsm

```diff
@@ -90,6 +92,10 @@ this.TopSitesFeed = class TopSitesFeed {
       this._currentSearchHostname = getShortURLForCurrentSearch();
       this.refresh({broadcast: true});
     }
+
+    if (topic === "browser-search-engine-modified") {
+      this.updateSearchShortcuts();
```

Comment:
^ Good enough or do I need to look at the topics too? @Mardak @k88hudson

Reply:
`s/topic/data/`

Reply:
Looks like we want "engine-added" and "engine-removed". There can be a lot of unrelated "engine-changed"

Reply:
Ah, we don't care about the user adding/removing custom engines as they won't have an identifier. And hiding a packaged one... can still show up in the list for now.